autopush/router/simple.py

```diff
@@ -142,14 +154,20 @@ def route_notification(self, notification, uaid_data):
                 router.clear_node,
                 uaid_data).addErrback(self._eat_db_err)
             self.metrics.increment("router.broadcast.miss")
-            returnValue(self.stored_response(notification))
+            result = yield maybeDeferred(self.stored_response, uaid,
+                                         notification)
+            returnValue(result)
 
         if result.code == 200:
             self.metrics.increment("router.broadcast.save_hit")
-            returnValue(self.delivered_response(notification))
+            result = yield maybeDeferred(self.delivered_response, uaid,
```

Comment:
`fernet.encrypt` shouldn't be throwing exceptions, but that's a risky position to take. We can generate the responses before routing or writing to storage, so we're guaranteed to have a valid URL to send back to the app server.